### ISSUE TITLE
[SofaGuitQt] High overhead when showing the profiler window

### ISSUE THREAD

**Issue by alxbilger** [sofa-framework/sofa#2142]
The Qt application is highly slowed down when the profiler window is displayed. The reasons:

- The tree is updated at each frame from scratch: 1) clearing the tree is time consuming, 2) building the widget tree is time consuming and 3) converting all the data (double) to `QString` is time consuming. I am not sure it is necessary to update the tree at each frame. We would need to find another mechanism (for example only when the button `Animate` is unpressed).
- The chart is updated at each frame from scratch

**Comment by epernod** [comment]
yes, I had a version where you impose the number of step from start so you don't swap the buffer. But you loose the nice "moving" display.
For me the idea of this tool is to debug so I'm not sure spending time on optimizing it is worth it. My assumption is the simulation is slower but the ratio of % duration for each step is the same. Maybe I'm wrong.

**Comment by alxbilger** [comment]
To be clear, the simulation is not slower. The measured durations don't have overhead. The time consuming part due to the GUI is not measured. So, you are right, the percentages are accurate, but the durations also. It just takes time to display them.
I agree that optimizing this tool is not a priority. This issue is to keep it in mind.
